prometheus_client/openmetrics/parser.py

```diff
@@ -106,11 +106,16 @@ def _parse_labels(it, text):
             elif char == '"':
                 if not core._METRIC_LABEL_NAME_RE.match(''.join(labelname)):
                     raise ValueError("Invalid line: " + text)
-                labels[''.join(labelname)] = ''.join(labelvalue)
+                utf8_str = ''.join(labelvalue)
+                try:
+                    utf8_str.encode('utf-8')
+                except:
```

Comment:
Catch the specific exception type that you're looking for

Reply:
@brian-brazil we're scratching our heads on finding a set of characters as a test case...

Reply:
Added UnicodeEncodeError exception type.